lexers/lexer.lua

```diff
@@ -1676,20 +1676,106 @@ M.ignored_extensions = {
 	orig = true, bak = true, old = true, new = true
 }
 
+--- Map of hashbang utilities to associated lexer names.
+-- @usage lexer.detect_utilities.luadoc = 'lua'
+M.detect_utilities = {}
```

Comment:
What is the difference between this and `M.detect_patterns`?

Reply:
POSIX calls utility the part of env which is the launcher, so in { env -S awk -f } awk would be the utility.

https://pubs.opengroup.org/onlinepubs/9699919799/utilities/env.html